### PR TITLE
refactor(helper): deprecate replaceSymbolWithNumber

### LINKED ISSUE #1994: Deprecate helpers.replaceSymbolWithNumber
### Clear and concise description of the problem

The method `helpers.replaceSymbolWithNumber` is mostly a black box that does some arbritary char -> digit transformations.

The implementation contains unexpected branches.

### Suggested solution

1. Replace all usages in our code base with a 

````ts
string.replace(/char+/g, (m) => faker.string.numeric(m.length));
````

Deprecate `helpers.replaceSymbolWithNumber`

### Alternative

Keep the method as is

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by suyashgulati**
Fixes #1994

Changes done:
- deprecated replaceSymbolWithNumber
- created a new legacy method legacyReplaceSymbolWithNumber
- methods using replaceSymbolWithNumber now use the legacy method instead

- no update on unit tests required

### CODE REVIEW

File: src/modules/helpers/index.ts [hunk 1]
```diff
@@ -158,6 +158,43 @@ function legacyRegexpStringParse(
...
+ * @param string The template string to parse. Defaults to `''`.
+ * @param symbol The symbol to replace with digits. Defaults to `'#'`.
+ *
+ * @example
+ * faker.helpers.legacyReplaceSymbolWithNumber(this.faker) // ''
+ * faker.helpers.legacyReplaceSymbolWithNumber(this.faker, '#####') // '04812'
+ * faker.helpers.legacyReplaceSymbolWithNumber(this.faker, '!####') // '27378'
+ * faker.helpers.legacyReplaceSymbolWithNumber(this.faker, 'Your pin is: !####') // '29841'
+ *
+ * @since 9.0.0
```
Comment (ST-DDT) [inline src/modules/helpers/index.ts]:
```suggestion
 * @since 8.4.0
```

File: src/modules/helpers/index.ts [hunk 2]
```diff
@@ -198,27 +235,26 @@ export class SimpleHelpersModule extends SimpleModuleBase {
...
+   *
    * @example
    * faker.helpers.replaceSymbolWithNumber() // ''
    * faker.helpers.replaceSymbolWithNumber('#####') // '04812'
    * faker.helpers.replaceSymbolWithNumber('!####') // '27378'
    * faker.helpers.replaceSymbolWithNumber('Your pin is: !####') // '29841'
    *
    * @since 2.0.1
+   *
+   * @deprecated Use `faker.string.numeric()` instead. Example: `string.replace(/#+/g, (m) => faker.string.numeric(m.length));`
```
Comment (ST-DDT) [inline src/modules/helpers/index.ts]: Maybe rename the variable to reduce conflicts with `faker.string`?

```suggestion
   * @deprecated Use `faker.string.numeric()` instead. Example: `value.replace(/#+/g, (m) => faker.string.numeric(m.length));`
```

File: src/modules/helpers/index.ts [hunk 3]
```diff
@@ -290,7 +326,7 @@ export class SimpleHelpersModule extends SimpleModuleBase {
   /**
    * Replaces the symbols and patterns in a credit card schema including Luhn checksum.
    *
-   * This method supports both range patterns `[4-9]` as well as the patterns used by `replaceSymbolWithNumber()`.
+   * This method supports both range patterns `[4-9]` as well as the patterns used by `legacyReplaceSymbolWithNumber()`.
```
Comment (ST-DDT) [inline src/modules/helpers/index.ts]: This is a reference to a method and thus should stick with the deprecated but public API method name.

File: src/modules/helpers/index.ts [hunk 4]
```diff
@@ -198,27 +235,26 @@ export class SimpleHelpersModule extends SimpleModuleBase {
...
-      } else if (string.charAt(i) === '!') {
-        str += this.faker.number.int({ min: 2, max: 9 });
-      } else {
-        str += string.charAt(i);
-      }
-    }
+    deprecated({
+      deprecated: 'faker.helpers.replaceSymbolWithNumber',
+      proposed: 'string.replace(/#+/g, (m) => faker.string.numeric(m.length))',
+      since: '9.0.0',
```
Comment (ST-DDT) [inline src/modules/helpers/index.ts]:
```suggestion
      since: '8.4',
      until: '9.0',
```

File: src/modules/helpers/index.ts [hunk 5]
```diff
@@ -158,6 +158,43 @@ function legacyRegexpStringParse(
   return string;
 }
 
+/**
+ * Parses the given string symbol by symbol and replaces the placeholders with digits (`0` - `9`).
+ * `!` will be replaced by digits >=2 (`2` - `9`).
+ *
+ * @internal
+ *
+ * @param faker Faker instance
```
Comment (ST-DDT) [inline src/modules/helpers/index.ts]:
```suggestion
 * @param faker A Faker instance.
```

Same as the method above.

File: src/modules/helpers/index.ts [hunk 6]
```diff
@@ -158,6 +158,43 @@ function legacyRegexpStringParse(
...
+ *
+ * @param faker A Faker instance.
+ * @param string The template string to parse. Defaults to `''`.
+ * @param symbol The symbol to replace with digits. Defaults to `'#'`.
+ *
+ * @example
+ * faker.helpers.legacyReplaceSymbolWithNumber(this.faker) // ''
+ * faker.helpers.legacyReplaceSymbolWithNumber(this.faker, '#####') // '04812'
+ * faker.helpers.legacyReplaceSymbolWithNumber(this.faker, '!####') // '27378'
+ * faker.helpers.legacyReplaceSymbolWithNumber(this.faker, 'Your pin is: !####') // '29841'
```
Comment (ST-DDT) [inline src/modules/helpers/index.ts]:
```suggestion
 * legacyReplaceSymbolWithNumber(faker) // ''
 * legacyReplaceSymbolWithNumber(faker, '#####') // '04812'
 * legacyReplaceSymbolWithNumber(faker, '!####') // '27378'
 * legacyReplaceSymbolWithNumber(faker, 'Your pin is: !####') // '29841'
```

**Review by ST-DDT — DISMISSED** [review]

File: src/modules/helpers/index.ts [hunk 7]
```diff
@@ -158,6 +158,43 @@ function legacyRegexpStringParse(
   return string;
 }
 
+/**
+ * Parses the given string symbol by symbol and replaces the placeholders with digits (`0` - `9`).
+ * `!` will be replaced by digits >=2 (`2` - `9`).
```
Comment (ST-DDT) [inline src/modules/helpers/index.ts:163]:
```suggestion
 * `!` will be replaced by digits >=2 (`2` - `9`).
 *
 * Note: This method will be removed in v9.
```

File: src/modules/helpers/index.ts [hunk 8]
```diff
@@ -158,6 +158,43 @@ function legacyRegexpStringParse(
...
+ *
+ * @internal
+ *
+ * @param faker A Faker instance.
```
Comment (ST-DDT) [inline src/modules/helpers/index.ts]: Sorry, for not suggesting the correct one in the first run.

```suggestion
 * @param faker The Faker instance to use.
```

File: src/modules/helpers/index.ts (hunk 7 above)
Comment (xDivisionByZerox) [reply src/modules/helpers/index.ts:163]: Can we put the note in a separate line? I think that makes more sense as it is a completly new information (doesn't refer to the sentences before).

Comment (ST-DDT) [reply src/modules/helpers/index.ts:163]: Sure

**Review by ST-DDT — APPROVED** [review]

**Review by xDivisionByZerox — APPROVED** [review]

**Pull request merged**